### DISCUSSION TITLE
How to use the different parts of observation data to train the NN at different epoch?

### DISCUSSION THREAD

**Discussion by Dingiso-cyber** [lululxvi/deepxde#952] [Q&A]
Hi Dr. Lu
    In the Residual domain, I can use this callback [dde.callbacks.PDEResidualResampler] function to reduce the occupied GPU memory. I don't know if there is a function like that for observation data in the inverse problems. My observation data is too big. How to use the different parts of observation data to train the NN at different epoch.  Thank you!

**Comment by Dingiso-cyber** [comment]
Maybe I can check the code of [dde.callbacks.PDEResidualResampler] and then write a new function for observation data. Is that possible?

**Reply by forxltk** [reply]
See #744 and #764.

**Reply by Dingiso-cyber** [reply]
Thank you, I will change the code tomorrow. That's really help me a lot

**Reply by Dingiso-cyber** [reply]
Hello, I am not sure if the batch_size in dde.icbc.PointSetBC must be used together with the dde.callbacks.PDEPointResampler. When I set the batch_size = 100 without the set the dde.callbacks.PDEPointResampler, my code doesn't work at all.

**Reply by forxltk** [reply]
https://github.com/lululxvi/deepxde/blob/master/examples/pinn_inverse/elliptic_inverse_field_batch.py

**Reply by Dingiso-cyber** [reply]
geomtime = dde.geometry.PointCloud(gen_point())
    obtf, ob_y = observation()
    observe_br = dde.icbc.PointSetBC(obtf, ob_y[:, 0:1], component=0,batch_size=3200)
    observe_sv = dde.icbc.PointSetBC(obtf, ob_y[:, 1:2], component=1,batch_size=3200)
    observe_sa = dde.icbc.PointSetBC(obtf, ob_y[:, 2:3], component=2,batch_size=3200)
    data = dde.data.PDE(
        geomtime,
        Induction_equation,
        [observe_br, observe_sv,observe_sa],
        anchors=obtf,
        num_domain=3200*N,
        num_test=5000,
    )
    net = dde.nn.PFNN([40] + [100,100,100,100,100] * 2 + [5], "tanh", "Glorot uniform")
    net.apply_feature_transform(feature_transform)
    resampler = dde.callbacks.PDEPointResampler()
    checkpointer = dde.callbacks.ModelCheckpoint(
        "model/model", verbose=2, save_better_only=True)
    model = dde.Model(data, net)
    model.compile("adam", lr=0.001, loss_weights=[50, 0, 10, 0, 100, 100, 100])
    losshistory, train_state = model.train(iterations=20000,callbacks=[checkpointer,resampler]) 

This is my code, my question is if I can remove the resampler callbacks. Thank you.

**Reply by lululxvi** [reply]
You cannot. `dde.callbacks.PDEPointResampler` does the real work of resampling.
